detectors/aws/ecs/ecs.go

```diff
@@ -43,25 +43,19 @@ var (
 )
 
 // Create interface for methods needing to be mocked
-type detectorUtils interface {
+type detectorUtilsResources interface {
 	getContainerName() (string, error)
 	getContainerID() (string, error)
 }
 
-// struct implements detectorUtils interface
-type ecsDetectorUtils struct{}
+// struct implements detectorUtilsResources interface
+type DetectorUtils struct{}
 
 // resource detector collects resource information from Elastic Container Service environment
 type ResourceDetector struct {
```

Comment:
Is there a reason that we should export `ResourceDetector`? Should we provide an unexported struct `resourceDetector ` and provide a function called this `NewResourceDetector`?

Reply:
:100:

Reply:
Good call. Added `NewResourceDetector` function.

Reply:
Just the linter is failing with this error `exported func NewResourceDetector returns unexported type go.opentelemetry.io/contrib/detectors/aws/ecs.resourceDetector, which can be annoying to use (golint)`

Reply:
Probably either we can export this struct or maybe we can use `exclude-rules` for this. I think we should export the struct. Any suggestions?

Reply:
@bhautikpip having `NewResourceDetector` return a [`resource.Detector`](https://github.com/open-telemetry/opentelemetry-go/blob/e50a1c8cecb8c584bdfada6cb21fe0b173a22646/sdk/resource/auto.go#L30-L38) instead of the unexported struct will solve this issue.

Reply:
Done @MrAlias